src/frequenz/client/dispatch/types.py

```diff
@@ -296,7 +296,11 @@ def from_protobuf(cls, pb_object: PBDispatch) -> "Dispatch":
             type=pb_object.data.type,
             create_time=to_datetime(pb_object.metadata.create_time),
             update_time=to_datetime(pb_object.metadata.modification_time),
-            end_time=to_datetime(pb_object.metadata.end_time),
+            end_time=(
+                to_datetime(pb_object.metadata.end_time)
+                if pb_object.metadata.HasField("end_time")
```

Comment:
I think we have this *bug* in **many** places in all API clients. We need to be more careful about optional/non-existing fields... For sure I've seen it in the microgrid API, but I think we are lucky there because most stuff we should check if they exist is always filled by the server, so we are not seeing the effect of the bug there.